ui/shared/components/panel/view-pedigree-image/PedigreeImagePanel.jsx

```diff
@@ -1,133 +1,348 @@
 import React from 'react'
-import { connect } from 'react-redux'
 import PropTypes from 'prop-types'
-import { Icon, Message, Segment } from 'semantic-ui-react'
+import { connect } from 'react-redux'
+import { ErrorBoundary } from 'react-error-boundary'
+import $ from 'jquery'
+import 'jquery-ui/ui/widgets/dialog'
+import 'jquery-ui/themes/base/all.css'
+import { Icon, Segment, Table } from 'semantic-ui-react'
 import styled from 'styled-components'
 
-import { updateFamily, RECEIVE_DATA } from 'redux/rootReducer'
-import { closeModal } from 'redux/utils/modalReducer'
-import DeleteButton from '../../buttons/DeleteButton'
+import { svg2img } from 'pedigreejs/es/io'
+import { current as currentDataset } from 'pedigreejs/es/pedcache'
+import {
+  build as buildPedigeeJs, rebuild as rebuildPedigeeJs, validate_pedigree as validatePedigree,
+} from 'pedigreejs/es/pedigree'
+import { copy_dataset as copyPedigreeDataset, messages as pedigreeMessages } from 'pedigreejs/es/pedigree_utils'
+
+import { getIndividualsByFamily } from 'redux/selectors'
+import { openModal } from 'redux/utils/modalReducer'
+import { INDIVIDUAL_FIELD_CONFIGS, INDIVIDUAL_FIELD_SEX, AFFECTED } from 'shared/utils/constants'
+import { snakecaseToTitlecase } from 'shared/utils/stringUtils'
+import ReduxFormWrapper from '../../form/ReduxFormWrapper'
+import { BooleanCheckbox, InlineToggle, RadioGroup, YearSelector } from '../../form/Inputs'
 import Modal from '../../modal/Modal'
-import { XHRUploaderWithEvents } from '../../form/XHRUploaderField'
-import { HorizontalSpacer } from '../../Spacers'
-import { ButtonLink } from '../../StyledComponents'
+import { NoBorderTable, FontAwesomeIconsContainer, ButtonLink } from '../../StyledComponents'
+import { EditPedigreeImageButton, DeletePedigreeImageButton, SavePedigreeDatasetButton } from './PedigreeImageButtons'
 
-const PedigreeImage = styled.img.attrs({ alt: 'pedigree' })`
-  max-height: ${props => (props.compact ? '35px' : '150px')};
+const PedigreeImg = styled.img.attrs({ alt: 'pedigree' })`
+  max-height: ${props => props.maxHeight}px;
   max-width: 225px;
   vertical-align: top;
   cursor: ${props => (props.disablePedigreeZoom ? 'auto' : 'zoom-in')};
 `
 
-class BaseEditPedigreeImageButton extends React.PureComponent {
+const PedigreeJsContainer = styled(FontAwesomeIconsContainer)`
+  display: ${props => ((props.disablePedigreeZoom && props.isEditable) ? 'auto' : 'none')};
+  cursor: ${props => (props.disablePedigreeZoom ? 'auto' : 'zoom-in')};
+
+  i.fa {
+    cursor: pointer;
+  }
+  
+  @font-face {
+    font-family: 'ped-icon-overrides';
```

Comment:
in general, we can map font-awesome classes to semantic ui unicode definitions in order to make them work with our icon library. However, there are 2 icons used by pedgree.js in a weird way such that its impossible to get them to render properly with the font library we have, so I added a small font library just for these 2 icons